src/main/java/com/marek/onlinebookstore/service/book/BookServiceImpl.java

```diff
@@ -18,58 +22,66 @@
+        Book book = bookMapping.toEntity(createBookRequestDto);
+        book.setCategories(createBookRequestDto.categoriesId().stream()
+                .map(categoryRepository::findById)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toSet())
+        );
+        return bookMapping.toDtoWithoutCategoryIds(bookRepository.save(book));
     }
 
     @Override
     public List<BookDto> findAll(Pageable pageable) {
-        return bookRepository.findAll(pageable)
-                .stream()
-                .map(bookMapping::toBookDto)
+        return bookRepository.findAll(pageable).stream()
+                .map(bookMapping::toDto)
                 .toList();
     }
 
     @Override
     public BookDto findById(Long id) {
-        return bookMapping.toBookDto(
-                bookRepository.findById(id)
-                        .orElseThrow(() -> new EntityNotFoundException(
-                                "Can't find book with id: " + id))
-        );
+        return bookMapping.toDto(
+                bookRepository.findById(id).orElseThrow(
+                        () -> new EntityNotFoundException("Can't find book with id: " + id)));
     }
 
     @Override
     public void deleteById(Long id) {
-        if (!bookRepository.existsById(id)) {
-            throw new EntityNotFoundException("Can't delete book with id: "
-                    + id + " because it does not exist.");
-        }
         bookRepository.deleteById(id);
     }
 
     @Override
-    public BookDto update(Long id, CreateBookRequestDto bookRequestDto) {
+    public BookDto update(Long id, CreateBookRequestDto createBookRequestDto) {
         if (!bookRepository.findById(id).isPresent()) {
             throw new EntityNotFoundException("Can't find book with id: " + id);
         }
-        Book book = bookMapping.toModel(bookRequestDto);
+        Book book = bookMapping.toEntity(createBookRequestDto);
         book.setId(id);
-        return bookMapping.toBookDto(bookRepository.save(book));
+        return bookMapping.toDto(bookRepository.save(book));
     }
 
     @Override
     public List<BookDto> searchBooks(BookSearchParametersDto searchParameters, Pageable pageable) {
         Specification<Book> bookSpecification = bookSpecificationBuilder.build(searchParameters);
         return bookRepository.findAll(bookSpecification, pageable)
-                .stream()
-                .map(bookMapping::toBookDto)
+                .stream().map(bookMapping::toDto).toList(); // Zwracamy List<BookDto>
+    }
+
+    @Override
+    public List<BookDtoWithoutCategoryIds> findByCategoryId(Long id, Pageable pageable) {
+        List<Book> allByCategoriesId = bookRepository.findAllByCategoriesId(id, pageable);
+        if (allByCategoriesId.isEmpty()) {
+            throw new EntityNotFoundException("Can't find books with category id: " + id);
+        }
```

Comment:
Maybe instead of throwing an exception you can just use the default approach (it will return an empty list), if so, just remove this if statement.